roles/edpm_nova/tasks/configure.yml

```diff
@@ -118,3 +118,35 @@
     mode: '0600'
     owner: nova
     group: nova
+
+- name: Check if compute_id exists in state_path
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.stat:
+    path: /var/lib/nova/compute_id
+  register: compute_id
+
+- name: Pre-generate compute_id if not already exists
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.shell:
+    cmd: uuidgen > "/var/lib/nova/compute_id"
```

Comment:
I had a bias against shell module, sorry :D 

But i guess you can replace two tasks with this one. What about:
```
    - name: Pre-generate compute_id if not already exists
      ansible.builtin.copy:
        dest: "/var/lib/nova/compute_id"
        owner: nova
        group: nova
        mode: "0400"
        attributes: "+i"
        content: |
          "{{ lookup('community.general.random_string') | to_uuid }}"
```

Reply:
hum honestly i think uuidgen is a better approach then that.
this can be the command module if you don't like shell

but you have made the ansible massively more complicated 

we also still need to keep the "name: Make compute_id ready-only"
later so setting the permission here is kind of redundant.

Reply:
> we also still need to keep the "name: Make compute_id ready-only"
later so setting the permission here is kind of redundant.

Yes, we need to keep the file permission settings independently as the file might exists during adoption but we want to still change its permissions.

Reply:
@SeanMooney i'm fine with `shell` module if it's unavoidable or makes the life easier.
(can't be `command` since you need to use redirection, which is unsupported on that module)

Reply:
`
    - name: Pre-generate compute_id if not already exists
      tags:
        - configure
        - nova
      become: true
      ansible.builtin.copy:
        dest: "/var/lib/nova/compute_id"
        owner: nova
        group: nova
        mode: "0400"
        attributes: "+i"
        content: |
          "{{ lookup('ansible.builtin.pipe', 'uuidgen') }}"
      when: not compute_id.stat.exists
`
how about this

Reply:
if only that doesn't introduce unexpected newlines and closing/wrapping lines